GitUI/CommandsDialogs/FormCleanupRepository.cs

```diff
@@ -78,17 +90,32 @@ private CleanMode GetCleanMode()
             throw new NotSupportedException($"Unknown value for {nameof(CleanMode)}.");
         }
 
-        private string? GetPathArgumentFromGui()
+        private string? GetInclusivePathArgumentFromGui()
         {
-            if (!checkBoxPathFilter.Checked)
+            if (!checkBoxIncludePathFilter.Checked)
             {
                 return null;
             }
 
             // 1. get all lines from text box which are not empty
             // 2. wrap lines with ""
             // 3. join together with space as separator
-            return string.Join(" ", textBoxPaths.Lines.Where(a => !string.IsNullOrEmpty(a)).Select(a => string.Format("\"{0}\"", a)));
+            return string.Join(" ", textBoxIncludePaths.Lines.Where(a => !string.IsNullOrEmpty(a)).Select(a => string.Format("\"{0}\"", a)));
+        }
+
+        private string? GetExclusivePathArgumentFromGui()
+        {
+            if (!checkBoxExcludePathFilter.Checked)
+            {
+                return null;
+            }
+
+            // 1. get all lines from text box which are not empty
+            // 2. Prepend lines with '-e /'
+            // 3. Replace whitespace with '?' and backslash with forward slash
+            // 3. join together with space as separator
+
+            return string.Join(" ", textBoxExcludePaths.Lines.Where(a => !string.IsNullOrEmpty(a)).Select(a => string.Format("-e /{0}", a.Replace(" ", "?").Replace("\\", "/"))));
```

Comment:
I don't think `.Replace("\\", "/")` should be used like this. Whilst it's not officially supported, the app can run on Linux, so all paths must be in the POSIX form. We have helpers for this specific purpose:
https://github.com/gitextensions/gitextensions/blob/8b5eab531dcc849d00110df6f9797a096e68b500/GitCommands/PathUtil.cs#L19-L24

Reply:
Good to know, I've implemented it.